LevelComponents/Entity.cpp

```diff
@@ -255,6 +255,100 @@ namespace LevelComponents
         return pixmap.toImage();
     }
 
+    /// <summary>
+    /// Render oam array in test mode.
+    /// </summary>
+    /// <param name="oamNum">
+    /// the number of oam.
+    /// </param>
+    /// <param name="nakedOAMarray">
+    /// Naked OAM array without oam number as the first element
+    /// </param>
+    /// <param name="noReferrneceBox">
+    /// Don't draw the yellow referrence Box if set this to true
+    /// </param>
+    QImage Entity::TestRenderOams(int oamNum, unsigned short *nakedOAMarray, bool noReferrneceBox)
+    {
+        if (!oamNum) return QImage();
+
+        QVector<OAMTile *> tmpOAMTiles;
+        for (int i = 0; i < oamNum; ++i)
+        {
+            // Obtain short values for the OAM tile
+            unsigned short attr0 = nakedOAMarray[3 * i];
+            unsigned short attr1 = nakedOAMarray[3 * i + 1];
+            unsigned short attr2 = nakedOAMarray[3 * i + 2];
+            // skip priority reset and Char Id (tiles ID) reset
+            // skip orientation change
+            // skip affine transformation (double size)
+            // skip alpha blend
+
+            // Obtain the tile parameters for the OAM tile
+            struct OAMTile *newOAM = new struct OAMTile();
```

Comment:
This is a memory leak. Change the vector to not use pointers

Reply:
it is deleted in the render for loop